### PR TITLE
add microsoft/bond 9.0.0

### PR DESCRIPTION
**Pull request by yelu**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/bond/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    exports_sources = ["FindBond.cmake"]
+
+    def source(self):
+        self.run("git clone https://github.com/microsoft/bond.git")
+        self.run("git checkout fe6f582ce4beb65644d9338536066e07d80a0289", cwd='bond')
+        self.run("git submodule update --init --recursive", cwd='bond')
+        tools.replace_in_file("bond/CMakeLists.txt",
+                              "project (bond)",
+                              '''project (bond)
+message("BOOST_ROOT=$ENV{BOOST_ROOT}")''')
```
Comment (madebr) [inline recipes/bond/all/conanfile.py:31]: Can you please package release 9.1: https://github.com/microsoft/bond/releases/tag/9.0.1
Then, you can use `conandata.yml` + export_source a `CMakeLists.txt` file, as all other recipes do.

File: recipes/bond/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    description = "Bond is a cross-platform framework for working with schematized data. It supports cross-language de/serialization and powerful generic mechanisms for efficiently manipulating data. Bond is broadly used at Microsoft in high scale services."
+    topics = ("bond", "microsoft")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [False, True],
+        "fPIC": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True}
+    build_requires = "boost/1.71.0@conan/stable"
```
Comment (madebr) [inline recipes/bond/all/conanfile.py:20]:
```suggestion
    requires = "boost/1.73..0"
```
A `require` is really necessary.
If boost is shared, then the library must be installed too.

File: recipes/bond/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        cmake = self.configure_cmake()
+        # CMAKE_INSTALL_PREFIX is set to self.package_folder
+        cmake.install()
+        if platform.system() == 'Linux':
+            self.run("chmod +rx *", cwd=f'{self.package_folder}/bin')
+        self.copy("FindBond.cmake", ".", ".")
+
+    def package_info(self):
+        self.cpp_info.includedirs = ['include']
+        self.cpp_info.libdirs = ['lib/bond']
```
Comment (madebr) [inline recipes/bond/all/conanfile.py:55]: Isn't the library installed in the `lib` folder?
```suggestion
        self.cpp_info.libdirs.append(os.path.join("lib", "bond"))
```

File: recipes/bond/all/test_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+conan_basic_setup()
+
+add_executable(example example.cpp)
+target_link_libraries(example ${CONAN_LIBS})
+
+# CTest is a testing tool that can be used to test your project.
+# enable_testing()
+# add_test(NAME example
+#          WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/bin
+#          COMMAND example)
```
Comment (madebr) [inline recipes/bond/all/test_package/CMakeLists.txt:14]:
```suggestion
target_link_libraries(example ${CONAN_LIBS})
```

File: recipes/bond/all/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class BondTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is
+        # in "test_package"
```
Comment (madebr) [inline recipes/bond/all/test_package/conanfile.py:13]:
```suggestion
```

File: recipes/bond/all/test_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
+            os.chdir("bin")
+            self.run(".%sexample" % os.sep)
```
Comment (madebr) [inline recipes/bond/all/test_package/conanfile.py:20]:
```suggestion
        if not tools.cross_building(self.settings):
            bin_path = os.path.join("bin", "example")
            self.run(bin_path, run_environment=True)
```

File: recipes/bond/all/test_package/example.cpp [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#include <iostream>
+#include <bond/core/bond_const_enum.h>
+
+int main() {
+    std::cout << "bond is installed successfully" << std::endl;
```
Comment (madebr) [inline recipes/bond/all/test_package/example.cpp:5]: Please call at least one bond function (not using a define or header-only function).
That way we can be sure the library is tested properly.

File: recipes/bond/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [False, True],
+        "fPIC": [True, False]}
+    default_options = {
+        "shared": False,
+        "fPIC": True}
+    build_requires = "boost/1.71.0@conan/stable"
+    generators = "cmake"
+    exports_sources = ["FindBond.cmake"]
```
Comment (madebr) [inline recipes/bond/all/conanfile.py:22]: Please remove this file (and export a conan wrapper `CMakeLists.txt` file instead).

As a consumer of this recipe, when using the `cmake_find_package` generator, conan will genere a `FindBond.cmake` script automatically.

File: recipes/bond/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+    def package(self):
+        cmake = self.configure_cmake()
+        # CMAKE_INSTALL_PREFIX is set to self.package_folder
+        cmake.install()
+        if platform.system() == 'Linux':
+            self.run("chmod +rx *", cwd=f'{self.package_folder}/bin')
+        self.copy("FindBond.cmake", ".", ".")
+
+    def package_info(self):
```
Comment (madebr) [inline recipes/bond/all/conanfile.py:53]: Apply this change only if bond creates `BondConfig.cmake` or `FindBond.cmake` files.
```suggestion
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "Bond"
        self.cpp_info.names["cmake_find_package_multi"] = "Bond"
```

File: recipes/bond/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+        self.run("git clone https://github.com/microsoft/bond.git")
+        self.run("git checkout fe6f582ce4beb65644d9338536066e07d80a0289", cwd='bond')
+        self.run("git submodule update --init --recursive", cwd='bond')
+        tools.replace_in_file("bond/CMakeLists.txt",
+                              "project (bond)",
+                              '''project (bond)
+message("BOOST_ROOT=$ENV{BOOST_ROOT}")''')
+
+    def configure_cmake(self):
+        if platform.system() == "Linux":
```
Comment (uilianries) [inline recipes/bond/all/conanfile.py:32]:
```suggestion
    def _configure_cmake(self):
        if tools.os_info.is_linux:
```

You can't add public methods, use protected instead.

File: recipes/bond/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+            self.run("curl -sSL https://get.haskellstack.org/ | sh", ignore_errors=True)
```
Comment (uilianries) [inline recipes/bond/all/conanfile.py:33]: Why do you need haskell? Downloading here is not allowed, but instead you can have a Conan package for haskell.

File: recipes/bond/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        cmake = self.configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self.configure_cmake()
+        # CMAKE_INSTALL_PREFIX is set to self.package_folder
+        cmake.install()
+        if platform.system() == 'Linux':
+            self.run("chmod +rx *", cwd=f'{self.package_folder}/bin')
+        self.copy("FindBond.cmake", ".", ".")
```
Comment (uilianries) [inline recipes/bond/all/conanfile.py:49]:
```suggestion
```

Not allowed. Use `cmake_find_package` generator instead.

File: recipes/bond/all/conanfile.py [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+    def build(self):
+        cmake = self.configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self.configure_cmake()
+        # CMAKE_INSTALL_PREFIX is set to self.package_folder
+        cmake.install()
+        if platform.system() == 'Linux':
+            self.run("chmod +rx *", cwd=f'{self.package_folder}/bin')
```
Comment (uilianries) [inline recipes/bond/all/conanfile.py:48]:
```suggestion
        if tools.os_info.is_linux:
            self.run("chmod +rx *", cwd=f'{self.package_folder}/bin')
```

Use os.chmod instead. Is there an executable here?

File: recipes/bond/all/conanfile.py [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+        self.copy("FindBond.cmake", ".", ".")
+
+    def package_info(self):
+        self.cpp_info.includedirs = ['include']
+        self.cpp_info.libdirs = ['lib/bond']
+        self.cpp_info.libs = ["bond"]
```
Comment (uilianries) [inline recipes/bond/all/conanfile.py:54]:
```suggestion
        self.cpp_info.libs = ["bond"]
        self.cpp_info.builddirs.append(os.path.join("lib", "cmake", "bond"))
        self.cpp_info.build_modules.append(os.path.join("lib", "cmake", "bond", "Bond.cmake"))
```

The cmake file offered by author contains some macros. Conan can't re-create it, so the best approach is copying it to your package.

https://github.com/microsoft/bond/blob/master/cmake/Bond.cmake

File: recipes/bond/all/conanfile.py [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        cmake = CMake(self)
+        cmake.definitions["BOND_ENABLE_GRPC"] = 'FALSE'
+        cmake.configure(source_folder="bond")
+        return cmake
+
+    def build(self):
+        cmake = self.configure_cmake()
+        cmake.build()
+
+    def package(self):
```
Comment (uilianries) [inline recipes/bond/all/conanfile.py:43]:
```suggestion
    def package(self):
        self.copy("LICENSE", dst="licenses", src="bond")
        self.copy("Bond.cmake", dst=os.path.join("lib", "cmake", "bond"), src=os.path.join("bond", "cmake))
```

Don't forget the license
Bond.cmake seems be useful.

**Pull request closed without merging**